servor.js

```diff
@@ -1,162 +1,238 @@
+      fs.stat(uri, (err, stat) => {
+        if (err) return sendError(res, resource, 404);
+        fs.readFile(uri, 'binary', (err, file) => {
+          if (err) return sendError(res, resource, 500);
+          if (isRoute && reload) file += reloader(inject);
+          sendFile(res, resource, status, file, ext);
+        });
+      });
+    }
+  }).listen(parseInt(port, 10));
 
-// ----------------------------------
-// Start static file server
-// ----------------------------------
+  // Notify livereload clients on file change
 
-http
-  .createServer((req, res) => {
-    const pathname = url.parse(req.url).pathname;
-    const isRoute = isRouteRequest(pathname);
-    const status = isRoute && pathname !== '/' ? 301 : 200;
-    const resource = isRoute ? `/${fallback}` : decodeURI(pathname);
-    const uri = path.join(cwd, root, resource);
-    const ext = uri.replace(/^.*[\.\/\\]/, '').toLowerCase();
-    isRoute && console.log('\n \x1b[44m', 'RELOADING', '\x1b[0m\n');
-    // Check if files exists at the location
-    fs.stat(uri, (err, stat) => {
-      if (err) return sendError(res, resource, 404);
-      // Respond with the contents of the file
-      fs.readFile(uri, 'binary', (err, file) => {
-        if (err) return sendError(res, resource, 500);
-        if (isRoute && reload) file += reloadScript;
-        sendFile(res, resource, status, file, ext);
-      });
+  reload &&
+    fs.watch(root, { recursive: true }, () => {
+      while (clients.length > 0)
+        sendMessage(clients.pop(), 'message', 'reload');
+      log();
     });
-  })
-  .listen(parseInt(port, 10));
-
-// ----------------------------------
-// Get available IP addresses
-// ----------------------------------
-const interfaces = require('os').networkInterfaces();
-const ips = Object.values(interfaces)
-  .reduce((a, b) => [...a, ...b], [])
-  .filter(ip => ip.family === 'IPv4' && ip.internal === false)
-  .map(ip => `http://${ip.address}:${port}`);
-
-// ----------------------------------
-// Log startup details to terminal
-// ----------------------------------
-
-console.log(`\n 🗂  Serving files from ./${root} on http://localhost:${port}`);
-ips.length > 0 && console.log(` 📡 Exposed to the network on ${ips[0]}`);
-console.log(` 🖥  Using ${fallback} as the fallback for route requests`);
-console.log(` ♻️  Reloading the browser when files under ./${root} change`);
-
-// ----------------------------------
-// Open the page in the default browser
-// ----------------------------------
-
-const page = `http://localhost:${port}`;
-const open =
-  process.platform == 'darwin'
-    ? 'open'
-    : process.platform == 'win32'
-    ? 'start'
-    : 'xdg-open';
 
-browser && require('child_process').exec(open + ' ' + page);
+  // Open the page in the default browse
+
+  browse && proc.execSync(`${open} ${protocol}://localhost:${port}`);
+
+  // Log state to the terminal
+
+  console.log('\n'.repeat(process.stdout.rows));
```

Comment:
Can we maybe skip this clear screen if `silent === true`?

Reply:
Ahh, good catch! Yes we can 👍